third_party/ngen/ngen.hpp

```diff
@@ -96,7 +97,13 @@ class BinaryCodeGenerator
     using RootCodeGenerator = BinaryCodeGenerator;
     static constexpr HW hardware = hw;
     static constexpr HW getHardware() { return hardware; }
-    void cancelAutoSWSB() { cancelAutoSWSB_ = true; }
+    void enableCancelAutoSWSB() {cancelAutoSWSB_.reset(new std::atomic<bool>(false));}
+    void cancelAutoSWSB() {
+#ifdef NGEN_SAFE
+        if(cancelAutoSWSB_ == nullptr) throw invalid_object_exception();
+#endif
+        if(cancelAutoSWSB_) *cancelAutoSWSB_ = true;
+    }
```

Comment:
Since this changes the default behaviour of `cancelAutoSWSB` from enabled to disabled, this requires changes to my benchstone refactoring PR. Please either add those changes to my existing gemmstone PR, or remind me to make those changes on that PR once this is merged to ngen.